mgnify_pipelines_toolkit/analysis/amplicon/primer_substrings.sh

```diff
@@ -0,0 +1,19 @@
+
+# Extract substrings to check for primer matching
```

Comment:
Same with this one... I would say to port to python code if we need to call it from python scripts

Reply:
Otherwise.. I think this should be an nf module (it's pretty specific for ASVs)

Reply:
Also replaced this with the new python fetch_mcp function I wrote - along with changes to the `standard_primer_matching.py` script that used to use `primer_substrings.sh` I was able to reproduce the exact same output as before for the unit tests